pkg/service/retryable_backup_handler.go

```diff
@@ -0,0 +1,76 @@
+package service
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/aerospike/backup-go/models"
+)
+
+type RetryableBackupHandler struct {
+	sync.RWMutex
+	handler BackupHandler
+	errCh   chan error
+}
+
+func newRetryableBackupHandler(
+	ctx context.Context,
+	retry *RetryService,
+	start func(ctx context.Context) (BackupHandler, error),
+	onFail func(ctx context.Context),
+	onSuccess func(ctx context.Context, stats *models.BackupStats) error,
+) *RetryableBackupHandler {
+	h := &RetryableBackupHandler{
+		errCh: make(chan error, 1),
+	}
+
+	processBackup := func() error {
+		handler, err := start(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to start backup: %w", err)
+		}
+
+		h.setHandler(handler)
+
+		if err = handler.Wait(ctx); err != nil {
+			onFail(ctx)
+			h.setHandler(nil)
+			return fmt.Errorf("backup failed: %w", err)
+		}
+
+		return onSuccess(ctx, handler.GetStats())
+	}
+
+	// Start the backup process with retries
+	go func() {
+		h.errCh <- retry.retry("backup", processBackup)
+	}()
```

Comment:
This routine initialization is unexpected in the constructor.

Reply:
agree, renamed the method, its not constructor anymore